Sync upstream 2022-03-16

nunnatsa opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Sync upstream 2022-03-16
```

ravidbro commented on pkg/managers/vm/machine_scope.go:
```diff
@@ -242,7 +242,9 @@ func (s *machineScope) buildVMITemplate(namespace string) (*kubevirtapiv1.Virtua
 
-	template.Spec = kubevirtapiv1.VirtualMachineInstanceSpec{}
+	template.Spec = kubevirtapiv1.VirtualMachineInstanceSpec{
+		TerminationGracePeriodSeconds: func(src int64) *int64 { return &src }(terminationGracePeriodSeconds),
```
Why do you need a function here?
I see that until now, the struct was created empty, and then all fields were updated on that. Why behave differently in that case?

nirarg replied:
That is way to pass a pointer in one liner
I wasn't sure if using it was clear, so because of your comment, I changed the value to be defined before

ravidbro commented on pkg/managers/vm/machine_scope.go:
```diff
@@ -46,10 +46,10 @@ const (
+	providerIDFormat                  = "kubevirt://%s/%s"
 )
 
-const providerIDFormat = "kubevirt://%s/%s"
```
I understand the change, but I think it should be in a different commit

nirarg replied:
Removed this change from the PR, will refactor later

chenyosef commented:
/lgtm

nirarg committed:
Change OWNERS file

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Automatic sync with upstream

This PR adds a new github action to daily sync with upstream. It checks if there are new changes in upstream, and if there are, it creates a PR to sync.

The new PR is created by hard reseting to the upstream main branch, then restoring the downstream modified files.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
add script to sync with upstream

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
create PR

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

0xFelix committed:
Fix typo in kubevirtci::fetch_kubevirtci

KUEVIRTCI_TAG --> KUBEVIRTCI_TAG

Signed-off-by: Felix Matouschek <fmatouschek@redhat.com>

k8s-ci-robot committed:
Merge pull request #112 from 0xFelix/patch-1

Fix typo in kubevirtci::fetch_kubevirtci

davidvossel commented:
/hold

we need release-1.0 sycned with main

nunnatsa commented:
> /hold
> 
> we need release-1.0 sycned with main

@davidvossel - I opened a new PR for that on #26
